builtin/logical/ssh/path_sign.go

```diff
@@ -511,10 +512,29 @@ func (b *creationBundle) sign() (retCert *ssh.Certificate, retErr error) {
 		},
 	}
 
+	sshAlgorithmSigner, _ := b.Signer.(ssh.AlgorithmSigner)
+
+	// prepare certificate for signing
+	certificate.Nonce = make([]byte, 32)
+	if _, err := io.ReadFull(rand.Reader, certificate.Nonce); err != nil {
+		return nil, fmt.Errorf("failed to generate signed SSH key")
+	}
+	certificate.SignatureKey = sshAlgorithmSigner.PublicKey()
+
+	// get bytes to sign
+	c2 := *certificate
+	c2.Signature = nil
+	out := c2.Marshal()
+	certificateBytes := out[:len(out)-4]
```

Comment:
Why are we stripping off the last 4 bytes? Maybe add a comment to clarify that?

Reply:
These bytes appear to hold the length of...something.

https://github.com/golang/crypto/blob/master/ssh/certs.go#L460

Reply:
It's the signature length.  This is based on some a private method in the go crypto/ssh lib (certificateBytes).